### ISSUE TITLE
Warping affected while handling Invalid gradients during training

### ISSUE THREAD

**Issue by VjayalakshmiK** [NagabhushanSN95/Pose-Warping#1]
Thanks for the repo!
I am using the torch version of your code in a training setup, to warp frame 1 into frame 2 and use it as supervision. 
During gradient computation, I run into an error at ```line 197 ``` due to division by zero- I think at ```cropped_weights```

```warped_frame2 = torch.where(mask, cropped_warped_frame / cropped_weights, zero_tensor)```

When I add a small residue like ```1e-10```, the warping using ground truth that was fine all along got affected- the warped frame is black. If I remove this residue and run the warping, I get the proper warped frame. 
So, adding this residue for training purpose is affecting the warping itself. 
How do I handle this?

**Comment by NagabhushanSN95** [comment]
Try 1e-12, that worked for me.

**Comment by NagabhushanSN95** [comment]
Closing this for inactivity. Please reopen if the problem is still unsolved.

**Comment by BoMingZhao** [comment]
> Try 1e-12, that worked for me.

I use 1e-12 still have the problem. Not only do NaN values appear during the backward, but the color of the image also seem off. Is there a solution to address this issue?

**Comment by NagabhushanSN95** [comment]
Try setting `torch.autograd.set_detect_anomaly(True)`. It'll show where exactly nan occurs.
